src/reflection/util/genutil.ts

```diff
@@ -77,29 +77,38 @@ for (const [scalarType, {type, literalKind}] of Object.entries(
 
 export function toTSScalarType(
   type: introspect.PrimitiveType,
-  types: introspect.Types
+  types: introspect.Types,
+  opts = {unionEnums: false, edgedbDatatypePrefix: "_."}
 ): CodeFragment[] {
   switch (type.kind) {
     case "scalar": {
       if (type.enum_values && type.enum_values.length) {
+        if (opts.unionEnums) {
+          return [`(${type.enum_values.map(quote).join(" | ")})`];
+        }
```

Comment:
Why union and not the native Typescript `enum` type?

Reply:
To mirror the type that the querybuilder would infer if you had `select`ed the property. I think there's also a concern with having a TS enum type, that because it exists as both a static type and a runtime object, it would be possible to confuse it with the other similar object that's generated for the enum (which represents the edgeql expression of the enum for use in building the queries).

Reply:
I think this should probably make these plain `enum`s - I don't think we have to worry deeply about the potential confusion surrounding multiple runtime exports, since the import will be a totally different file. Also, TS treats enums as assignable to equivalent string unions: 

```ts
enum Genre {
  Action = "Action",
  Horror = "Horror",
  Romcom = "Romcom"
}

const gen: "Action" | "Horror" | "Romcom" = Genre.Horror;
```